app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/monitor.rb

```diff
@@ -0,0 +1,41 @@
+class ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Monitor
+  def initialize(ems)
+    @ems               = ems
+    @alerts_client     = ems.alerts_connect
+    @collecting_events = false
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  def each_batch
+    while @collecting_events
+      yield fetch
+    end
+  end
+
+  private
+
+  # Each fetch is performed from the time of the most recently caught event or 1 minute back for the first poll.
+  # This gives us some slack if hawkular events are timestamped behind the miq server time.
+  # TODO: Is there a danger here?  This assumes all Hawkular events at time T are fetched in one call.
+  # TODO: Can we store start_time and resume where we left off?  We miss events when we're not running...
```

Comment:
I would remove this TODO. The other providers are doing it the same and I dont think we can store that time somewhere.